pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -45,10 +46,17 @@ type isImmutableFunc func(runtime.Object) bool
 type objectCacheItem struct {
 	refCount  int
 	store     cache.Store
+	reflector *cache.Reflector
+
 	hasSynced func() (bool, error)
 
+	lastAccessTime time.Time
+	immutable      bool
+	stopped        bool
+	initialized    *bool
+
 	// lock is protecting from closing stopCh multiple times.
-	lock   sync.Mutex
+	lock   *sync.RWMutex
```

Comment:
In general, RWMutex is more expensive than Mutex.

Given that we don't have any non-negligible contention here, let's revert to Mutex and use Lock/Unlock everywhere.